### DISCUSSION TITLE
Are replacements still limited to npm as mentioned in README.md?

### DISCUSSION THREAD

**Discussion by pdecat** [renovatebot/renovate#24277] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.87.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Not an issue, really a question.

The main `README.md` file mentions:

> - Get replacement PRs to migrate from a deprecated dependency to the community suggested replacement (*npm packages only*)

See https://github.com/renovatebot/renovate/blame/36.87.1/readme.md#L27

Is this still true?

Documentation seems to indicate otherwise, as presets with other datasources like `nuget` and `docker` exist, e.g.:
* https://docs.renovatebot.com/presets-replacements/#replacementsk8s-registry-move
* https://docs.renovatebot.com/presets-replacements/#replacementsspectre-cli-to-spectre-console-cli

Bonus question: How hard would it be to add a replacement preset for `asdf` to replace a CLI tool by another?

### What you tried so far.

Nothing yet.

### Relevant debug logs

<details><summary>Logs</summary>

```
N/A
```

</details>

**Comment by rarkins** [comment]
Related: https://github.com/renovatebot/renovate/issues/14149

**Reply by pdecat** [reply]
Issue https://github.com/renovatebot/renovate/issues/14149 seems to mention the managers which are not supported by replacement.

So, isn't the home page statement "_npm packages only_" outdated given `docker` and `nuget` are supported?

**Reply by rarkins** [reply]
Yes. PR welcome

**Reply by pdecat** [reply]
Also, I see https://github.com/renovatebot/renovate/pull/17883 was merged, are all managers actually supported?

**Reply by rarkins** [reply]
@t-kulmburg @viceice do you recall what our exact status is regarding replacements?

**Reply by pdecat** [reply]
Submitted https://github.com/renovatebot/renovate/pull/24293 in case all managers are now covered.

**Reply by viceice** [reply]
not all are covered.

**Reply by viceice** [reply]
- https://github.com/renovatebot/renovate/issues/14149

**Reply by pdecat** [reply]
So are those in the https://github.com/renovatebot/renovate/issues/14149 issue description the only ones not supported?

That's what the last comment seems to indicate: https://github.com/renovatebot/renovate/issues/14149#issuecomment-1409822662

**Reply by viceice** [reply, marked as answer]
yes

**Reply by pdecat** [reply]
Thanks, I'll update my PR accordingly!
